tests/config/test_component_factory.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+from typing import Dict, Union
+
+from omegaconf import OmegaConf
+
+from modalities.config.component_factory import ComponentFactory
+from tests.config.hierarchical_instantiation.configs import (
+    CompVConfig,
+    CompWConfig,
+    CompXConfig,
+    CompYConfig,
+    CompZConfig,
+    ReferenceConfig,
+)
+
+
+def load_app_config_dict(config_file_path: Path) -> Dict:
+    cfg = OmegaConf.load(config_file_path)
+    config_dict = OmegaConf.to_container(cfg, resolve=True)
+    return config_dict
```

Comment:
This function also exists in `src/modalities/__main__.py`. The only difference seems to be logging. If that is a problem, I would suggest to add an argument `verbose: bool = True` to the function, which skips the logging if `verbose` is `False`.

Reply:
fixed. kept only the one in __main__.py